cpu/sam4s/cpu.c

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2025 Mesotic SAS
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_sam4s
+ * @{
+ *
+ * @file
+ * @brief       Implementation of the CPU initialization
+ *
+ * @author      Dylan Laduranty <dylan.laduranty@mesotic.com>
+ * @}
+ */
+
+#include "cpu.h"
+#include "board.h"
+#include "kernel_init.h"
+#include "periph_conf.h"
+#include "periph/init.h"
+#include "stdio_base.h"
+
+#define XTAL_STARTUP        (8U)
+#define PLL_CNT             (64U)
+
+#if ((CLOCK_PLL_MUL < 7) || (CLOCK_PLL_MUL > 62))
+#error "CLOCK_PLL_MUL has an incorrect value"
+#endif
+
+/**
+ * @brief Initialize the CPU, set IRQ priorities
+ */
+void cpu_init(void)
+{
+    /* SAM4s MCUs requires WPKEY for enabling peripheral */
+    PMC->PMC_WPMR = PMC_WPMR_WPKEY_PASSWD;
+
+    /* disable the watchdog timer */
+    WDT->WDT_MR |= WDT_MR_WDDIS;
+
+    /* initialize the Cortex-M core */
+    cortexm_init();
+
+    /* setup the flash wait states */
+    EFC0->EEFC_FMR = EEFC_FMR_FWS(CLOCK_FWS);
+    EFC1->EEFC_FMR = EEFC_FMR_FWS(CLOCK_FWS);
+
+    /* enable the Cortex-M Cache Controller */
+    CMCC->CMCC_CTRL |= CMCC_CTRL_CEN;
+
+    /* unlock write protect register for PMC module */
+    PMC->PMC_WPMR = PMC_WPMR_WPKEY_PASSWD;
+
+    /* activate the external crystal */
+    PMC->CKGR_MOR = (CKGR_MOR_KEY_PASSWD |
+                     CKGR_MOR_MOSCXTST(XTAL_STARTUP) |
+                     CKGR_MOR_MOSCXTEN |
+                     CKGR_MOR_MOSCRCEN);
```

Comment:
This is more a personal taste thing, but don't you agree that this looks much better?

```suggestion
    PMC->CKGR_MOR = CKGR_MOR_KEY_PASSWD
                  | CKGR_MOR_MOSCXTST(XTAL_STARTUP)
                  | CKGR_MOR_MOSCXTEN
                  | CKGR_MOR_MOSCRCEN;
```

Reply:
This is heavily borrow from `cpu/sam3/cpu.c` so I guess the same thing applied there.
But I should have run `uncrustify` or `clangd-format` here tbh.

I'd like to cleanup this a bit later, once we support more older Atmel MCUs to see how much we can share across them. SAMG55 will be probably my next target.

Anyway, you're right I'll fix your comments in the morning.